app/components/version_milestones_component.rb

```diff
@@ -15,12 +15,28 @@ def steps
     @steps ||= milestones_presenter.steps_for(version)
   end
 
-  def current_version?
-    version.to_i == milestones_presenter.current_version
+  def link_user_version?
+    return false if version.to_i == milestones_presenter.current_version.to_i
+    return false if user_version == user_versions_presenter.user_version.to_i
+
+    true
+  end
+
+  def link_to_document?
+    version.to_i == milestones_presenter.current_version.to_i \
+     && user_version == user_versions_presenter.head_user_version&.to_i \
+     && user_versions_presenter.user_version.present?
+  end
+
+  def user_version_link
+    return link_to(user_version_label, user_version_path) if link_user_version?
+    return link_to(user_version_label, solr_document_path(milestones_presenter.druid)) if link_to_document?
+
+    user_version_label
   end
```

Comment:
It looks like this can return either a link or a label. Worth considering a rename to `user_version_link_or_label` or some such?

Reply:
Good suggestion. Fixed.